webrtc.html

```diff
@@ -3985,46 +4080,66 @@ <h4 id="sec.create-identity-proxy">Instantiating an IdP Proxy</h4>
+          <p>The <code>RTCPeerConnection</code> instantiates an IdP as described
+          in <a href="#sec.identity-proxy-communications"></a> and <a
+          href="#sec.register-idp"></a>.  If the IdP cannot be loaded,
+          instantiated or the IdP proxy is not registered, this process
+          fails.</p>
         </li>
 
-        <li>The <code>RTCPeerConnection</code> sends a "SIGN" message to the IdP
-        proxy. This message includes the material
-        the <code>RTCPeerConnection</code> desires to be bound to the user's
-        identity.</li>
-
-        <li>If the user has been authenticated by the IdP, and the IdP is
-        willing to generate an identity assertion, the IdP generates an
-        identity assertion. This step depends entirely on the IdP. The methods
-        by which an IdP authenticates users or generates assertions is not
-        specified, though this could involve interacting with the IdP server or
-        other servers.</li>
+        <li>
+          <p>The <code>RTCPeerConnection</code> invokes the <code><a
+          href="#widl-RTCIdentityProvider-generateAssertion-Promise-RTCIdentityAssertionResult--DOMString-contents-DOMString-origin-DOMString-usernameHint">generateAssertion</a></code>
+          method on the <code><a>RTCIdentityProvider</a></code> instance
+          registered by the IdP.</p>
+
+          <p>The <code>RTCPeerConnection</code> generates the
+          <var>contents</var> parameter to this method as described in
+          [[!RTCWEB-SECURITY-ARCH]].  The <var>origin</var> parameter contains
+          the origin of the script that calls the <code>RTCPeerConnection</code>
+          method that triggers this behavior.  The <var>usernameHint</var> value
+          is the same value that is provided to
+          <code>setIdentityProvider</code>, if any such value was provided.</p>
+        </li>
 
-        <li>The IdP proxy sends a response containing the identity assertion to
-        the <code>RTCPeerConnection</code> over the message channel.</li>
+        <li>
+          <p>The IdP returns a Promise to the <code>RTCPeerConnection</code>.
+          If the user has been authenticated by the IdP, and the IdP is willing
+          to generate an identity assertion, the IdP resolves the promise with
+          an identity assertion in the form of an
+          <code><a>RTCIdentityAssertionResult</a></code>.</p>
+
+          <p>This step depends entirely on the IdP. The methods by which an IdP
+          authenticates users or generates assertions is not specified, though
+          this could involve interacting with the IdP server or other
+          servers.</p>
+        </li>
 
-        <li>The <code>RTCPeerConnection</code> MAY store the identity assertion
-        for use with future offers or answers.</li>
+        <li>
+          <p>The <code>RTCPeerConnection</code> MAY store the identity assertion
+          for use with future offers or answers.</p>
+        </li>
```

Comment:
Hmm... it's not clear how to know if it can be reused. Minimally you need the same key, but what if the IdP has anti-replay stuff? Seems like more is needed here.